### ISSUE TITLE
Ops that need to support Dynamic Shape

### ISSUE THREAD

**Issue by vanbasten23** [pytorch/xla#4756]
Below is a list of ops that need special upgrade for supporting Dynamic Shape in Mask RCNN. They come from the [dynamic shape design doc](https://docs.google.com/document/d/1qUBwlLbDPsWKP3kaSCEgIF4hf-C4vvpA10upEvZep_s/edit?pli=1#). Here we break the ops into several logical categories that highlight what degrees of support will be needed for the ops.

The checked ops are the ones that we have verified to support dynamic shape.

1. Ops that generate dynamism
- [x] torch.nonzero
- [x] torch.unique
- [ ] torch_xla.nms

2. Ops Falling Back to aten in PyTorch/XLA

The ops below are, at some level, not supported in PyTorch/XLA. To support them we either need lowering work or dynamic shape API support work.

- [ ] aten::_local_scalar_dense
- [ ] aten::randperm.generator_out
  We may not need to lower this op because the input to the op is an integer.
- [ ] aten::upsample_bilinear2d
- [ ] aten::upsample_nearest2d.vec

3. Nativefunction Ops Supported by PyTorch/XLA
The ops below are lowered in PyTorch/XLA and execute on XLA devices as expected. However, as will be discussed later, some of these ops may require additional support for Dynamic Shape integration. They need the additional support if (a) they cannot propagate dynamic information due to the format of their inputs (e.g. `torch.expand` takes `size` as input and `size` doesn’t propagate dynamic information), or (b) they are composite ops that utilize dynamic ops under the hood at lowering (e.g. `torch.fill_` uses `torch.expand` under the hood).

- [ ] xla::_copy_from
- [ ] xla::_copy_from_and_resize
- [ ] xla::_log_softmax
- [ ] xla::_s_where
- [ ] xla::_softmax
- [ ] xla::_to_cpu
- [ ] xla::add
- [ ] xla::addmm
- [ ] xla::arange_out
- [ ] xla::binary_cross_entropy_with_logits
- [ ] xla::bitwise_and
- [ ] xla::bitwise_or_out
- [ ] xla::cat
- [ ] xla::clamp
- [ ] xla::clamp_min
- [ ] xla::clone
- [ ] xla::convolution_overrideable
- [ ] xla::div
- [x] xla::empty (https://github.com/pytorch/xla/pull/4550)
- [ ] xla::eq
- [ ] xla::exp
- [x] xla::expand
- [ ] xla::fill_
- [ ] xla::floor
- [ ] xla::ge
- [ ] xla::gt
- [ ] xla::index
- [ ] xla::index_put_
- [ ] xla::log
- [ ] xla::log2
- [ ] xla::lt
- [ ] xla::max
- [ ] xla::max_pool2d
- [ ] xla::maximum
- [ ] xla::mean
- [ ] xla::minimum
- [ ] xla::mul
- [ ] xla::neg
- [ ] xla::nll_loss_forward
- [ ] xla::permute
- [ ] xla::relu
- [ ] xla::relu_
- [ ] xla::rsqrt
- [ ] xla::select
- [ ] xla::sigmoid
- [ ] xla::slice
- [ ] xla::smooth_l1_loss
- [ ] xla::split_with_sizes
- [ ] xla::sqrt
- [ ] xla::stack
- [ ] xla::sub
- [x] xla::t (https://github.com/pytorch/xla/pull/4606)
- [x] xla::topk
- [ ] xla::unbind
- [x] xla::unsqueeze (https://github.com/pytorch/xla/pull/4608)
- [ ] xla::upsample_bilinear2d
- [ ] xla::upsample_nearest2d
- [x] xla::view (https://github.com/pytorch/xla/pull/4629)
- [ ] xla::zero_

cc @miladm @JackCaoG @ymwangg

**Comment by miladm** [comment]
CC @ymwangg
